rdr_service/dao/participant_dao.py

```diff
@@ -268,8 +269,23 @@ def _do_update(self, session, obj, existing_obj):
                     obj.organizationId = None
                     need_new_summary = True
 
-        # No pairing updates sent, keep existing values.
-        if update_pairing == False:
+        if update_pairing:
+            if obj.organizationId != existing_obj.organizationId and existing_obj.participantSummary is not None:
+                # Get valid files ready for sync when a participant is paired to an organization
+                ConsentDao.set_previously_synced_files_as_ready(session, obj.participantId)
```

Comment:
Do we want to be able to quickly confirm which organizations should have gotten the files in their bucket?  I'm wondering if we need to implement a `sync_history` that we add to every time we move a file to SYNC_COMPLETE.  (I was thinking of the BYOD status JSON struct we have now in `participant_summary`)

If so, I don't know if there is a need to implement it in conjunction with this change, so as not to lose prior sync date(/organization?)  details that would be updated/overwritten.

Reply:
I don't think there's a pressing need for knowing where files had previously been copied to. Any usual investigation is for trying to determine if it copied to the bucket for the organization that needs to have it now. And the usual case this code will be handling is for participants that consented before the validation pipeline started and are being paired now.

Reply:
Per offline conversation, do not expect enough re-syncs (to repaired orgs) such that losing the previous `sync_time` value is an issue.   Per Kenny, only really interested in the most recent sync operation?